### PR TITLE
Restored Deencapsulation and its related classes.

### PR DESCRIPTION
**Pull request by yukkes**
The Deencapsulation class has been restored.
Added DeencapsulationTest class and have verified that all tests PASS.

```
[INFO] Results:
[WARNING] Tests run: 1047, Failures: 0, Errors: 0, Skipped: 10
```

### CODE REVIEW

**Comment by hazendaz** [comment]
Thanks @yukkes , will look to merge probably tomorrow and try to get a release out shortly after.

**Comment by yukkes** [comment]
Thank you for your response.
I restored the Deencapsulation.invoke() method earlier, so please check it.

File: main/src/main/java/mockit/internal/reflection/MethodReflection.java [hunk 1]
```diff
@@ -7,11 +7,7 @@
 import static java.lang.reflect.Modifier.isPrivate;
 import static java.lang.reflect.Modifier.isStatic;
 
-import static mockit.internal.reflection.ParameterReflection.acceptsArgumentTypes;
-import static mockit.internal.reflection.ParameterReflection.getParameterTypesDescription;
-import static mockit.internal.reflection.ParameterReflection.hasMoreSpecificTypes;
-import static mockit.internal.reflection.ParameterReflection.indexOfFirstRealParameter;
-import static mockit.internal.reflection.ParameterReflection.matchesParameterTypes;
+import static mockit.internal.reflection.ParameterReflection.*;
```
Comment (hazendaz) [inline main/src/main/java/mockit/internal/reflection/MethodReflection.java:10]: @yukkes Please don't use star imports.  Usually your IDE causes that, make sure intellij (going on a limb that is what you are using) is setup appropriately to stop doing that.  Its meant for use if something like 99 imports which isn't the case.

**Comment by hazendaz** [comment]
ok just fix the star import issue, rest I think is great.  I can merge this then and release right after.  If you need it to work with java 8 will need to know that as I will have to backport it.  If its fine at java 11, great makes my life easier :)

**Comment by hazendaz** [comment]
Actually I'm a bit impatient tonight.  I'm going to merge, I'll fix my trivial concern and release this bit!

**Pull request merged**